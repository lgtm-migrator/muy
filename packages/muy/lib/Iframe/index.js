import React from "react"
import makeStyles from "@material-ui/core/styles/makeStyles"
import { objectFitCover } from "@muy/styles"
import Box from "@material-ui/core/Box"

const useStyles = makeStyles(theme => ({
  cover: objectFitCover,
}))

const Iframe = ({ title, variant, ...props }) => {
  const classes = useStyles()
  return (
    <Box
      border={0}
      className={variant === "cover" && classes.cover}
      component="iframe"
      display="inline"
      frameBorder={0}
      height={variant === "cover" && "100%"}
      mb={0}
      mr={0}
      mt={0}
      p={0}
      title={title}
      width={variant === "cover" && "100%"}
      {...props}
    />
  )
}

export default Iframe
