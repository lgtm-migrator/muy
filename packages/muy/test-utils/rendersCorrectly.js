import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
