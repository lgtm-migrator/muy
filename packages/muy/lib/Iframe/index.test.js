import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import Iframe from "."

describe("<Iframe/> component", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<Iframe />)
  })
  it("should render correctly with cover variant", () => {
    rendersCorrectly(<Iframe variant={"cover"} />)
  })
})
