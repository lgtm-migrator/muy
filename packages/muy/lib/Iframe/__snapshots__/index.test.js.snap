// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Iframe/> component should render correctly with cover variant 1`] = `
<iframe
  className="MuiBox-root MuiBox-root-3 makeStyles-cover-1"
  frameBorder={0}
/>
`;

exports[`<Iframe/> component should render correctly with no props 1`] = `
<iframe
  className="MuiBox-root MuiBox-root-2"
  frameBorder={0}
/>
`;
